app/com/gu/floodgate/contentsource/ContentSource.scala

```diff
@@ -16,13 +22,28 @@ case class ContentSourcesResponse(contentSources: Seq[ContentSource])
 case class SingleContentSourceResponse(contentSource: ContentSource)
 
 @JsonFormat
-case class ContentSource(id: String,
+case class ContentSource(
+    id: String,
     appName: String,
     description: String,
     reindexEndpoint: String,
-    environment: String) {
+    environment: String,
+    authType: String) {
 
   def uniqueId: String = s"$id-$environment"
+
+  def reindexEndpointWithDateParams(dateParameters: DateParameters): String = {
+
+    def param(key: String, value: Option[String]): Option[String] = value.map(v => s"$key=$v")
+
+    val paramFrom = param("from", dateParameters.from map (_.toString))
+    val paramTo = param("to", dateParameters.to map (_.toString))
+    val queryString: String = Seq(paramFrom, paramTo).flatten.mkString("&")
+    val urlSeparator = if ("api-key" == this.authType) "&" else "?"
```

Comment:
I would change this predicate to `if (reindexEndpoint contains "?")`. The URL might already contain parameters for some reason unrelated to the api key.